lib/iris/analysis/__init__.py

```diff
@@ -1472,6 +1473,64 @@ def _proportion(array, function, axis, **kwargs):
     return result
 
 
+def _lazy_max_run(array, axis=-1, **kwargs):
+    """
+    Runs fairly clearly need the axis along which they're being calculated not
+    to be chunked. This wrapper uses
+    :meth:`~iris._lazy_data.map_complete_blocks` to guarantee that.
+    """
+
+    _partial_max_run_calc = functools.partial(
+        _lazy_max_run_calc,
+        axis=axis,
+        **kwargs,
+    )
+
+    result = iris._lazy_data.map_complete_blocks(
+        array, _partial_max_run_calc, (axis,), (0,)
+    )
+
+    # Check whether to reduce to a scalar result, as per the behaviour
+    # of other aggregators.
+    if result.shape == (1,):
+        result = da.squeeze(result)
+
+    return result
+
+
+def _lazy_max_run_calc(array, axis=-1, **kwargs):
+    """
+    Lazily perform the calculation of maximum run lengths along the given axis
+    """
+    array = iris._lazy_data.as_lazy_data(array)
+    func = kwargs.pop("function", None)
+    if not callable(func):
+        emsg = "function must be a callable. Got {}."
+        raise TypeError(emsg.format(type(func)))
+    bool_array = func(array)
+    bool_array = np.logical_and(
+        bool_array, np.logical_not(da.ma.getmaskarray(array))
+    )
+    padding = [(0, 0)] * array.ndim
+    padding[axis] = (0, 1)
+    ones_zeros = da.pad(bool_array, padding).astype(int)
```

Comment:
In order to get masked points to end a run, I think you'd have to put something like this here:
`ones_zeros = da.ma.filled(ones_zeros, 0)`

Reply:
I think unmasking the array earlier in the code makes sense.